src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -319,13 +357,11 @@ void LinearIR::LoopManager::update_loop_port(size_t loop_id, const ExpressionPor
         return;
 
     // to save other parameters except expression port
-    std::vector<LoopPort> target_loop_ports(target_ports.size(), *port_it);
-    std::transform(target_loop_ports.begin(), target_loop_ports.end(), target_ports.begin(), target_loop_ports.begin(),
-                   [](LoopPort loop_port, const ExpressionPort& expr_port) {
-                       LoopPort copy = std::move(loop_port);  // to save loop port parameters
-                       copy.expr_port = std::make_shared<ExpressionPort>(expr_port);
-                       return copy;
-                   });
+    std::vector<LoopPort> target_loop_ports;
+    target_loop_ports.reserve(target_ports.size());
+    for (const auto& p : target_ports)
+        target_loop_ports.push_back(*port_it->clone_with_new_expr(p.get_expr()));
+
```

Comment:
Please note that we need to change not only expression but port_idx too. See more detailed explanations in the [discussion](https://github.com/openvinotoolkit/openvino/pull/20242#discussion_r1354963045)

Reply:
Agreed. Good job!